velox/connectors/hive/HiveDataSource.cpp

```diff
@@ -579,62 +579,49 @@ void HiveDataSource::addSplit(std::shared_ptr<ConnectorSplit> split) {
   }
 
   auto& fileType = reader_->rowType();
+  std::vector<TypePtr> fileColumnTypes = fileType->children();
 
-  std::vector<TypePtr> columnTypes = fileType->children();
-  for (int i = 0; i < readerOutputType_->size(); i++) {
-    auto fieldName = readerOutputType_->nameOf(i);
-    auto scanChildSpec = scanSpec_->childByName(fieldName);
```

Comment:
A question: does the old code need to do nullable check here? Or we assume scan spec has one spec for each reader output child type? Thanks!

Reply:
> A question: does the old code need to do nullable check here? Or we assume scan spec has one spec for each reader output child type? Thanks!

Yes scan spec has one spec for each reader output child type. See HiveDataSource::makeScanSpec()
```
// Process columns that will be projected out.
  for (int i = 0; i < rowType->size(); ++i) {
    auto& name = rowType->nameOf(i);
    auto& type = rowType->childAt(i);
    auto it = outputSubfields.find(name);
    if (it == outputSubfields.end()) {
      spec->addFieldRecursively(name, *type, i);
      filterSubfields.erase(name);
      continue;
    }
```